### PR TITLE
lanraragi: init at 0.8.90

### LINKED ISSUE #247103: Package request: lanraragi
**Project description**

**Metadata**

* homepage URL: https://lrr.tvc-16.science/
* source URL: https://github.com/Difegue/LANraragi
* license: mit
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/247103

This PR adds 1 main package: `lanraragi`
it also adds 20 `perl` packages
there is also a `nixos` module which sets up a `systemd` service for LANraragi and another for the redis server being used.

### Notes

---

To run the package standalone (without the `nixos` module) you'll need to create `lrr.conf`, which may be copied from the GitHub repo.

With the `nixos` module, I initialize `lrr.conf` with `cat` in preparation for being able to securely set the Redis password in the future.

I have not yet made many `nixos` modules, so I am not sure if this is correct.

---

#### `pkgs/development/perl-modules/Alien-FFI-dont-download.patch`
this is so that it doesn't try to download `libffi` from GitHub
#### `pkgs/development/perl-modules/ArchiveLibarchive-set-findlib-path.patch`
`FFI::CheckLib`'s `find_lib_or_die` can either take an `Alien` or a `libpath`.
Originally `Archive::Libarchive` used an `Alien`, but that was too much of a hassle, so I instead specified the `libpath`, which also works. The patch also removes a version dependeny (`FFI:CheckLib` had a breaking change when specifying an `Alien`, but it doesn't matter if we use `libpath`)

#### `pkgs/by-name/la/lanraragi/fix-paths.patch`
LANraragi has a lot of absolute and relative path choices that are not acceptable in `/nix/store`.
There are some things (themes, error images, etc.) that should be relative to the install directory, but aren't.
There are other things (logs, temps) that should be relative to the working directory, but aren't.
Some of these could have been overwritten with env-vars, such as `LRR_TEMP_DIRECTORY`, but I believe it's better to have a working default without env-vars.

#### `pkgs/by-name/la/lanraragi/install.patch`
Originally, the install-script would check whether dependencies are installed, then it runs `npm install` and then copies the used assets. I removed the non-perl dependency checks (some were broken), and the `npm install` part. The script still checks whether we have all the perl dependencies with `App::cpanminus`.
This patch also lowers the `URI` minimum version, as `URI` has **A LOT** of packages depending on it, and updating it would be crazy in this PR.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by TomaSajt** [comment]
If you got pinged, sorry, it's probably because of the `Mojolicious` update. It rebuilt a lot of packages.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>182 packages built:</summary>
  <ul>
    <li>abcde</li>
    <li>convos</li>
    <li>convos.devdoc</li>
    <li>get_iplayer</li>
    <li>get_iplayer.man</li>
    <li>lanraragi</li>
    <li>perl536Packages.AlienFFI</li>
    <li>perl536Packages.AlienFFI.devdoc</li>
    <li>perl536Packages.ArchiveLibarchive</li>
    <li>perl536Packages.ArchiveLibarchive.devdoc</li>
    <li>perl536Packages.ArchiveLibarchiveExtract</li>
    <li>perl536Packages.ArchiveLibarchiveExtract.devdoc</li>
    <li>perl536Packages.ArchiveLibarchivePeek</li>
    <li>perl536Packages.ArchiveLibarchivePeek.devdoc</li>
    <li>perl536Packages.FFIC</li>
    <li>perl536Packages.FFIC.devdoc</li>
    <li>perl536Packages.FFICStat</li>
    <li>perl536Packages.FFICStat.devdoc</li>
    <li>perl536Packages.FFIPlatypus</li>
    <li>perl536Packages.FFIPlatypus.devdoc</li>
    <li>perl536Packages.FFIPlatypusTypeEnum</li>
    <li>perl536Packages.FFIPlatypusTypeEnum.devdoc</li>
    <li>perl536Packages.FFIPlatypusTypePtrObject</li>
    <li>perl536Packages.FFIPlatypusTypePtrObject.devdoc</li>
    <li>perl536Packages.FileShareDirDist</li>
    <li>perl536Packages.FileShareDirDist.devdoc</li>
    <li>perl536Packages.JSONValidator</li>
    <li>perl536Packages.JSONValidator.devdoc</li>
    <li>perl536Packages.LinkEmbedder</li>
    <li>perl536Packages.LinkEmbedder.devdoc</li>
    <li>perl536Packages.LogfileRotate</li>
    <li>perl536Packages.LogfileRotate.devdoc</li>
    <li>perl536Packages.Minion</li>
    <li>perl536Packages.Minion.devdoc</li>
    <li>perl536Packages.MinionBackendRedis</li>
    <li>perl536Packages.MinionBackendRedis.devdoc</li>
    <li>perl536Packages.MinionBackendSQLite</li>
    <li>perl536Packages.MinionBackendSQLite.devdoc</li>
    <li>perl536Packages.MinionBackendmysql</li>
    <li>perl536Packages.MinionBackendmysql.devdoc</li>
    <li>perl536Packages.MojoIOLoopDelay</li>
    <li>perl536Packages.MojoIOLoopDelay.devdoc</li>
    <li>perl536Packages.MojoIOLoopForkCall</li>
    <li>perl536Packages.MojoIOLoopForkCall.devdoc</li>
    <li>perl536Packages.MojoJWT</li>
    <li>perl536Packages.MojoJWT.devdoc</li>
    <li>perl536Packages.MojoPg</li>
    <li>perl536Packages.MojoPg.devdoc</li>
    <li>perl536Packages.MojoRedis</li>
    <li>perl536Packages.MojoRedis.devdoc</li>
    <li>perl536Packages.MojoSAML</li>
    <li>perl536Packages.MojoSAML.devdoc</li>
    <li>perl536Packages.MojoSQLite</li>
    <li>perl536Packages.MojoSQLite.devdoc</li>
    <li>perl536Packages.MojoUserAgentCached</li>
    <li>perl536Packages.MojoUserAgentCached.devdoc</li>
    <li>perl536Packages.Mojolicious</li>
    <li>perl536Packages.Mojolicious.devdoc</li>
    <li>perl536Packages.MojoliciousPluginAssetPack</li>
    <li>perl536Packages.MojoliciousPluginAssetPack.devdoc</li>
    <li>perl536Packages.MojoliciousPluginGravatar</li>
    <li>perl536Packages.MojoliciousPluginGravatar.devdoc</li>
    <li>perl536Packages.MojoliciousPluginMail</li>
    <li>perl536Packages.MojoliciousPluginMail.devdoc</li>
    <li>perl536Packages.MojoliciousPluginOpenAPI</li>
    <li>perl536Packages.MojoliciousPluginOpenAPI.devdoc</li>
    <li>perl536Packages.MojoliciousPluginRenderFile</li>
    <li>perl536Packages.MojoliciousPluginRenderFile.devdoc</li>
    <li>perl536Packages.MojoliciousPluginStatus</li>
    <li>perl536Packages.MojoliciousPluginStatus.devdoc</li>
    <li>perl536Packages.MojoliciousPluginSyslog</li>
    <li>perl536Packages.MojoliciousPluginSyslog.devdoc</li>
    <li>perl536Packages.MojoliciousPluginTemplateToolkit</li>
    <li>perl536Packages.MojoliciousPluginTemplateToolkit.devdoc</li>
    <li>perl536Packages.MojoliciousPluginTextExceptions</li>
    <li>perl536Packages.MojoliciousPluginTextExceptions.devdoc</li>
    <li>perl536Packages.MojoliciousPluginWebpack</li>
    <li>perl536Packages.MojoliciousPluginWebpack.devdoc</li>
    <li>perl536Packages.Mojomysql</li>
    <li>perl536Packages.Mojomysql.devdoc</li>
    <li>perl536Packages.MusicBrainz</li>
    <li>perl536Packages.MusicBrainz.devdoc</li>
    <li>perl536Packages.NetDNSNative</li>
    <li>perl536Packages.NetDNSNative.devdoc</li>
    <li>perl536Packages.OpenAPIClient</li>
    <li>perl536Packages.OpenAPIClient.devdoc</li>
    <li>perl536Packages.ParallelLoops</li>
    <li>perl536Packages.ParallelLoops.devdoc</li>
    <li>perl536Packages.Test2ToolsFFI</li>
    <li>perl536Packages.Test2ToolsFFI.devdoc</li>
    <li>perl536Packages.Test2ToolsMemoryCycle</li>
    <li>perl536Packages.Test2ToolsMemoryCycle.devdoc</li>
    <li>perl536Packages.TestArchiveLibarchive</li>
    <li>perl536Packages.TestArchiveLibarchive.devdoc</li>
    <li>perl538Packages.AlienFFI</li>
    <li>perl538Packages.AlienFFI.devdoc</li>
    <li>perl538Packages.ArchiveLibarchive</li>
    <li>perl538Packages.ArchiveLibarchive.devdoc</li>
    <li>perl538Packages.ArchiveLibarchiveExtract</li>
    <li>perl538Packages.ArchiveLibarchiveExtract.devdoc</li>
    <li>perl538Packages.ArchiveLibarchivePeek</li>
    <li>perl538Packages.ArchiveLibarchivePeek.devdoc</li>
    <li>perl538Packages.FFIC</li>
    <li>perl538Packages.FFIC.devdoc</li>
    <li>perl538Packages.FFICStat</li>
    <li>perl538Packages.FFICStat.devdoc</li>
    <li>perl538Packages.FFIPlatypus</li>
    <li>perl538Packages.FFIPlatypus.devdoc</li>
    <li>perl538Packages.FFIPlatypusTypeEnum</li>
    <li>perl538Packages.FFIPlatypusTypeEnum.devdoc</li>
    <li>perl538Packages.FFIPlatypusTypePtrObject</li>
    <li>perl538Packages.FFIPlatypusTypePtrObject.devdoc</li>
    <li>perl538Packages.FileShareDirDist</li>
    <li>perl538Packages.FileShareDirDist.devdoc</li>
    <li>perl538Packages.JSONValidator</li>
    <li>perl538Packages.JSONValidator.devdoc</li>
    <li>perl538Packages.LinkEmbedder</li>
    <li>perl538Packages.LinkEmbedder.devdoc</li>
    <li>perl538Packages.LogfileRotate</li>
    <li>perl538Packages.LogfileRotate.devdoc</li>
    <li>perl538Packages.Minion</li>
    <li>perl538Packages.Minion.devdoc</li>
    <li>perl538Packages.MinionBackendRedis</li>
    <li>perl538Packages.MinionBackendRedis.devdoc</li>
    <li>perl538Packages.MinionBackendSQLite</li>
    <li>perl538Packages.MinionBackendSQLite.devdoc</li>
    <li>perl538Packages.MinionBackendmysql</li>
    <li>perl538Packages.MinionBackendmysql.devdoc</li>
    <li>perl538Packages.MojoIOLoopDelay</li>
    <li>perl538Packages.MojoIOLoopDelay.devdoc</li>
    <li>perl538Packages.MojoIOLoopForkCall</li>
    <li>perl538Packages.MojoIOLoopForkCall.devdoc</li>
    <li>perl538Packages.MojoJWT</li>
    <li>perl538Packages.MojoJWT.devdoc</li>
    <li>perl538Packages.MojoPg</li>
    <li>perl538Packages.MojoPg.devdoc</li>
    <li>perl538Packages.MojoRedis</li>
    <li>perl538Packages.MojoRedis.devdoc</li>
    <li>perl538Packages.MojoSAML</li>
    <li>perl538Packages.MojoSAML.devdoc</li>
    <li>perl538Packages.MojoSQLite</li>
    <li>perl538Packages.MojoSQLite.devdoc</li>
    <li>perl538Packages.MojoUserAgentCached</li>
    <li>perl538Packages.MojoUserAgentCached.devdoc</li>
    <li>perl538Packages.Mojolicious</li>
    <li>perl538Packages.Mojolicious.devdoc</li>
    <li>perl538Packages.MojoliciousPluginAssetPack</li>
    <li>perl538Packages.MojoliciousPluginAssetPack.devdoc</li>
    <li>perl538Packages.MojoliciousPluginGravatar</li>
    <li>perl538Packages.MojoliciousPluginGravatar.devdoc</li>
    <li>perl538Packages.MojoliciousPluginMail</li>
    <li>perl538Packages.MojoliciousPluginMail.devdoc</li>
    <li>perl538Packages.MojoliciousPluginOpenAPI</li>
    <li>perl538Packages.MojoliciousPluginOpenAPI.devdoc</li>
    <li>perl538Packages.MojoliciousPluginRenderFile</li>
    <li>perl538Packages.MojoliciousPluginRenderFile.devdoc</li>
    <li>perl538Packages.MojoliciousPluginStatus</li>
    <li>perl538Packages.MojoliciousPluginStatus.devdoc</li>
    <li>perl538Packages.MojoliciousPluginSyslog</li>
    <li>perl538Packages.MojoliciousPluginSyslog.devdoc</li>
    <li>perl538Packages.MojoliciousPluginTemplateToolkit</li>
    <li>perl538Packages.MojoliciousPluginTemplateToolkit.devdoc</li>
    <li>perl538Packages.MojoliciousPluginTextExceptions</li>
    <li>perl538Packages.MojoliciousPluginTextExceptions.devdoc</li>
    <li>perl538Packages.MojoliciousPluginWebpack</li>
    <li>perl538Packages.MojoliciousPluginWebpack.devdoc</li>
    <li>perl538Packages.Mojomysql</li>
    <li>perl538Packages.Mojomysql.devdoc</li>
    <li>perl538Packages.MusicBrainz</li>
    <li>perl538Packages.MusicBrainz.devdoc</li>
    <li>perl538Packages.NetDNSNative</li>
    <li>perl538Packages.NetDNSNative.devdoc</li>
    <li>perl538Packages.OpenAPIClient</li>
    <li>perl538Packages.OpenAPIClient.devdoc</li>
    <li>perl538Packages.ParallelLoops</li>
    <li>perl538Packages.ParallelLoops.devdoc</li>
    <li>perl538Packages.Test2ToolsFFI</li>
    <li>perl538Packages.Test2ToolsFFI.devdoc</li>
    <li>perl538Packages.Test2ToolsMemoryCycle</li>
    <li>perl538Packages.Test2ToolsMemoryCycle.devdoc</li>
    <li>perl538Packages.TestArchiveLibarchive</li>
    <li>perl538Packages.TestArchiveLibarchive.devdoc</li>
  </ul>
</details>

**Review by h7x4 — COMMENTED** [review]
Only had a look at the module, don't know much about perl packages, so I'll leave that for other reviewers.

**Comment by TomaSajt** [comment]
I am not able to use a computer for a few days, I'll try to do what I can once I get back home

**Comment by h7x4** [comment]
No worries. This is all volunteer work after all :grin:

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 255715` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lanraragi</li>
  </ul>
</details>
<details>
  <summary>76 packages built:</summary>
  <ul>
    <li>perl536Packages.AlienFFI</li>
    <li>perl536Packages.AlienFFI.devdoc</li>
    <li>perl536Packages.ArchiveLibarchive</li>
    <li>perl536Packages.ArchiveLibarchive.devdoc</li>
    <li>perl536Packages.ArchiveLibarchiveExtract</li>
    <li>perl536Packages.ArchiveLibarchiveExtract.devdoc</li>
    <li>perl536Packages.ArchiveLibarchivePeek</li>
    <li>perl536Packages.ArchiveLibarchivePeek.devdoc</li>
    <li>perl536Packages.FFIC</li>
    <li>perl536Packages.FFIC.devdoc</li>
    <li>perl536Packages.FFICStat</li>
    <li>perl536Packages.FFICStat.devdoc</li>
    <li>perl536Packages.FFIPlatypus</li>
    <li>perl536Packages.FFIPlatypus.devdoc</li>
    <li>perl536Packages.FFIPlatypusTypeEnum</li>
    <li>perl536Packages.FFIPlatypusTypeEnum.devdoc</li>
    <li>perl536Packages.FFIPlatypusTypePtrObject</li>
    <li>perl536Packages.FFIPlatypusTypePtrObject.devdoc</li>
    <li>perl536Packages.FileShareDirDist</li>
    <li>perl536Packages.FileShareDirDist.devdoc</li>
    <li>perl536Packages.LogfileRotate</li>
    <li>perl536Packages.LogfileRotate.devdoc</li>
    <li>perl536Packages.MinionBackendRedis</li>
    <li>perl536Packages.MinionBackendRedis.devdoc</li>
    <li>perl536Packages.MojoliciousPluginRenderFile</li>
    <li>perl536Packages.MojoliciousPluginRenderFile.devdoc</li>
    <li>perl536Packages.MojoliciousPluginTemplateToolkit</li>
    <li>perl536Packages.MojoliciousPluginTemplateToolkit.devdoc</li>
    <li>perl536Packages.NetDNSNative</li>
    <li>perl536Packages.NetDNSNative.devdoc</li>
    <li>perl536Packages.ParallelLoops</li>
    <li>perl536Packages.ParallelLoops.devdoc</li>
    <li>perl536Packages.Test2ToolsFFI</li>
    <li>perl536Packages.Test2ToolsFFI.devdoc</li>
    <li>perl536Packages.Test2ToolsMemoryCycle</li>
    <li>perl536Packages.Test2ToolsMemoryCycle.devdoc</li>
    <li>perl536Packages.TestArchiveLibarchive</li>
    <li>perl536Packages.TestArchiveLibarchive.devdoc</li>
    <li>perl538Packages.AlienFFI</li>
    <li>perl538Packages.AlienFFI.devdoc</li>
    <li>perl538Packages.ArchiveLibarchive</li>
    <li>perl538Packages.ArchiveLibarchive.devdoc</li>
    <li>perl538Packages.ArchiveLibarchiveExtract</li>
    <li>perl538Packages.ArchiveLibarchiveExtract.devdoc</li>
    <li>perl538Packages.ArchiveLibarchivePeek</li>
    <li>perl538Packages.ArchiveLibarchivePeek.devdoc</li>
    <li>perl538Packages.FFIC</li>
    <li>perl538Packages.FFIC.devdoc</li>
    <li>perl538Packages.FFICStat</li>
    <li>perl538Packages.FFICStat.devdoc</li>
    <li>perl538Packages.FFIPlatypus</li>
    <li>perl538Packages.FFIPlatypus.devdoc</li>
    <li>perl538Packages.FFIPlatypusTypeEnum</li>
    <li>perl538Packages.FFIPlatypusTypeEnum.devdoc</li>
    <li>perl538Packages.FFIPlatypusTypePtrObject</li>
    <li>perl538Packages.FFIPlatypusTypePtrObject.devdoc</li>
    <li>perl538Packages.FileShareDirDist</li>
    <li>perl538Packages.FileShareDirDist.devdoc</li>
    <li>perl538Packages.LogfileRotate</li>
    <li>perl538Packages.LogfileRotate.devdoc</li>
    <li>perl538Packages.MinionBackendRedis</li>
    <li>perl538Packages.MinionBackendRedis.devdoc</li>
    <li>perl538Packages.MojoliciousPluginRenderFile</li>
    <li>perl538Packages.MojoliciousPluginRenderFile.devdoc</li>
    <li>perl538Packages.MojoliciousPluginTemplateToolkit</li>
    <li>perl538Packages.MojoliciousPluginTemplateToolkit.devdoc</li>
    <li>perl538Packages.NetDNSNative</li>
    <li>perl538Packages.NetDNSNative.devdoc</li>
    <li>perl538Packages.ParallelLoops</li>
    <li>perl538Packages.ParallelLoops.devdoc</li>
    <li>perl538Packages.Test2ToolsFFI</li>
    <li>perl538Packages.Test2ToolsFFI.devdoc</li>
    <li>perl538Packages.Test2ToolsMemoryCycle</li>
    <li>perl538Packages.Test2ToolsMemoryCycle.devdoc</li>
    <li>perl538Packages.TestArchiveLibarchive</li>
    <li>perl538Packages.TestArchiveLibarchive.devdoc</li>
  </ul>
</details>

File: pkgs/servers/lanraragi/cpucount.patch [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+-use Sys::CpuAffinity;
+ use Storable qw(lock_store);
+ use Mojo::JSON qw(to_json);
+ 
+@@ -147,7 +146,7 @@ sub update_filemap {
+     $redis->quit();
+ 
+     # Now that we have all new files, process them...with multithreading!
+-    my $numCpus = Sys::CpuAffinity::getNumCpus();
++    my $numCpus = 1;
```
[the remark this answers is not in the document]
Comment (TomaSajt) [reply pkgs/servers/lanraragi/cpucount.patch]: It turns out the issue was that Sys::CpuAffinity was checking every supported OS's capabilities in checkPhase, so I turned it checkPhase off, and now it works

File: nixos/modules/services/web-apps/lanraragi.nix [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        type = lib.types.str;
+        default = "lanraragi";
+        description = lib.mdDoc "User account under which LANraragi runs.";
+      };
+
+      group = lib.mkOption {
+        type = lib.types.str;
+        default = "lanraragi";
+        description = lib.mdDoc "Group account under which LANraragi runs.";
+      };
```
[the remark this answers is not in the document]
Comment (TomaSajt) [reply nixos/modules/services/web-apps/lanraragi.nix]: I updated the service to use DynamicUser, it seems to still work OK.

File: nixos/modules/services/web-apps/lanraragi.nix [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    users.groups.lanraragi = lib.mkIf (cfg.group == "lanraragi") { };
+
+    services.redis.servers.lanraragi = {
+      enable = true;
+      port = cfg.redis.port;
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
```
[the remark this answers is not in the document]
Comment (TomaSajt) [reply nixos/modules/services/web-apps/lanraragi.nix]: I removed the ability to have a custom dataDir, as I am not sure how to do this with DynamicUser. (also, it's not that important IMO)

**Comment by TomaSajt** [comment]
Currently, the admin password cannot be set from the nixos module, that will require patching the app a bit.

**Comment by TomaSajt** [comment]
Now the password can be set straight from the nixos module.
If not set, it will not be managed by the module but there will still be a password.
There is a no-password mode in the app, maybe if the password was left unset, it could enable that.
Anyways I think this is good enough for now.

**Review by stigtsp — COMMENTED** [review]
The new `perlPackages`, including patches to `lanraragi` and perl modules LGTM. 

- The new `perlPackages` additions should work fine with the pending perl-updates PR #262304
- I have not tested the application or reviewed `services.lanraragi`.

Suggestion: Not a requirement, but it would be nice with some basic nixos test for this service as well.

**Comment by TomaSajt** [comment]
I added a nixos test that checks whether the service starts correctly, and then checks whether it can log in as admin correctly.

**Comment by TomaSajt** [comment]
Also, upstream has added `package-lock.json` to version control, so there's no longer any need to have it commited here. (Now it is added with `fetchpatch`)

**Review by stigtsp — APPROVED** [review]
LGTM, thanks! Consider adding a release note to rl-2311.section.md

**Pull request merged**